packages/api-fetch/src/index.js

```diff
@@ -43,17 +43,32 @@ const DEFAULT_OPTIONS = {
 	credentials: 'include',
 };
 
+/**
+ * @type {import('./types').ApiFetchMiddleware[]}
+ */
 const middlewares = [
 	userLocaleMiddleware,
 	namespaceEndpointMiddleware,
 	httpV1Middleware,
 	fetchAllMiddleware,
 ];
 
+/**
+ * Register a middleware
+ *
+ * @param {import('./types').ApiFetchMiddleware} middleware
+ */
 function registerMiddleware( middleware ) {
 	middlewares.unshift( middleware );
 }
 
+/**
+ * Checks the status of a response, throwing the Response as an error if
+ * it is outside the 200 range.
+ *
+ * @param {Response} response
+ * @return {Response} The response if the status is OK.
```

Comment:
FWIW `OK` means just `200`. However we'll throw if it's any non-successful response (e.g. anything that's not with response code 2**)